infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/wsplugins/KubernetesPluginsToolingApplier.java

```diff
@@ -124,8 +128,16 @@ public void apply(
 
       Collection<CommandImpl> pluginRelatedCommands = commandsResolver.resolve(chePlugin);
 
+      DevfileImpl devfile = k8sEnv.getDevfile();
+      Optional<ComponentImpl> pluginRelatedComponentOpt = devfile.getComponents().stream()
+          .filter(c -> chePlugin.getId().equals(c.getId())).findAny();
```

Comment:
What about updating the ChePlugin model to include a `source` field, where we could set either plugin ID or reference as a unique identifier in the plugin broker (I can implement this side if it helps). If the Che server is going to start downloading plugin metas it may as well do all the work in the broker.

Reply:
it does not change a lot but makes a things a bit cleaner. Added the corresponding TODO